### PR TITLE
Ethashproof: specify custom cache location

### PR DESCRIPTION
**Pull request by rigidus**

### CODE REVIEW

File: eth2near/ethashproof/cmd/cache/main.go [hunk 1]
```diff
@@ -8,23 +8,50 @@ import (
...
+		usage("Error: Epoch number param is missing\n")
+	case 2:
+		epoch = getEpoch(epochIdx)
+		cacheDir = "|default|"
+	case 3:
+		epoch = getEpoch(epochIdx)
+		cacheDir = getCacheDir(cacheDirIdx)
 	}
+	// fmt.Printf("Epoch: %d\n", epoch)
+	// fmt.Printf("CacheDir: %s\n", cacheDir)
```
Comment (sept-en) [inline eth2near/ethashproof/cmd/cache/main.go:52]: Commented out code should be removed.

```suggestion
```

File: eth2near/ethashproof/cmd/cache/main.go [hunk 2]
```diff
@@ -8,23 +8,48 @@ import (
...
+
+	switch parcnt {
+	case 1:
+		usage("Error: Epoch number param is missing\n")
+	case 2:
+		epoch = getEpoch(epochIdx)
+		cacheDir = "|default|"
+	case 3:
+		epoch = getEpoch(epochIdx)
+		cacheDir = getCacheDir(cacheDirIdx)
```
Comment (sept-en) [inline eth2near/ethashproof/cmd/cache/main.go:49]:
```suggestion
	argsCount := len(os.Args)
	epochArgIdx := 1
	cacheDirArgIdx := 2
        var epoch int
	var cacheDir string

	switch argsCount {
	case 1:
		usage("Error: Epoch number param is missing\n")
	case 2:
		epoch = getEpoch(epochArgIdx)
		cacheDir = "|default|"
	case 3:
		epoch = getEpoch(epochArgIdx)
		cacheDir = getCacheDir(cacheDirArgIdx)
```

File: eth2near/ethashproof/dag_merkle_root.go [hunk 3]
```diff
@@ -46,10 +46,16 @@ func processDuringRead(f *os.File, startIn128Res int, fullSizeIn128Res uint32, m
 // 3. If saveCache is true, save root merkle tree of 10 levels
 //    to disk
 // 4. Return merkle root
-func CalculateDatasetMerkleRoot(epoch uint64, saveCache bool) (mtree.Hash, error) {
+func CalculateDatasetMerkleRoot(epoch uint64, saveCache bool, cacheDir string) (mtree.Hash, error) {
 	blockno := epoch * 30000
 	fmt.Printf("Make the dag\n")
-	ethash.MakeDAG(blockno, ethash.DefaultDir)
+
+	if cacheDir == "|default|" {
```
Comment (karim-en) [inline eth2near/ethashproof/dag_merkle_root.go:53]: Please don't use string literal as an option, you can just check for empty string here.

**Review by karim-en — CHANGES_REQUESTED** [review]

**Review by sept-en — CHANGES_REQUESTED** [review]
Closed as a stalled. Any new changes should take into account the existence of eth2-to-near-relay and eth2-client.

**Pull request closed without merging**